gobblin-runtime/src/main/java/org/apache/gobblin/runtime/kafka/HighLevelConsumer.java

```diff
@@ -197,6 +201,22 @@ protected void shutdownMetrics() throws IOException {
    */
   protected void createMetrics() {
     this.messagesRead = this.metricContext.counter(RuntimeMetrics.GOBBLIN_KAFKA_HIGH_LEVEL_CONSUMER_MESSAGES_READ);
+    this.queueSizeGauges = new ContextAwareGauge[numThreads];
+    this.queueSizeGaugeValues = new AtomicIntegerArray(numThreads);
+    for (int i=0; i < numThreads; i++) {
+      int finalI = i;
+      this.queueSizeGauges[i] = this.metricContext.newContextAwareGauge(
+          RuntimeMetrics.GOBBLIN_KAFKA_HIGH_LEVEL_CONSUMER_QUEUE_SIZE + "-" + i,
+          () -> queueSizeGaugeValues.get(finalI));
+    }
+  }
+
+  /**
+   * Update atomic integer corresponding to queue size
+   * @param queueIndex used to index the queue list
+   */
+  private void updateQueueSizes(int queueIndex) {
+    this.queueSizeGaugeValues.set(queueIndex, queues[queueIndex].size());
```

Comment:
since `AtomicIntegerArray::set` is not actually an atomic operation (of the kind `getAndIncrement` or even `getAndSet`), do we truly need such an instance?